test/unit_test/app_common/test_storage.py

```diff
@@ -45,50 +46,58 @@ def random_meta():
     return {random.getrandbits(8): random.getrandbits(8) for _ in range(32)}
 
 
-tmp_dir = tempfile.TemporaryDirectory()
-tmp_dir_name = tmp_dir.name
+tmp_dir = None
+storages = defaultdict()
+minio_ps = None
 
 
-@classmethod
-def teardown_class(cls):
-    tmp_dir.cleanup()
+@pytest.mark.xdist_group(name="storage_tests_group")
+@pytest.mark.parametrize("storage", [FilesystemStorage, S3Storage])
+class TestStorage:
 
+    @classmethod
+    def setup_class(cls):
+        global tmp_dir
+        global minio_ps
```

Comment:
Can we make `tmp_dir`, `minio_ps`, `storages` class variables?
I see they are only being used in the class.